### PR TITLE
Upgrade azure v12

### PR DESCRIPTION
**Pull request by rintcius**
[ch11156]

### CODE REVIEW

File: azure/src/main/scala/quasar/blobstore/azure/AzureListService.scala [hunk 1]
```diff
@@ -25,23 +25,25 @@ import java.lang.Integer
...
+
+  def apply[F[_]: ConcurrentEffect: ContextShift](
       toListBlobsOption: Kleisli[F, PrefixPath, ListBlobsOptions],
       mkArgs: ListBlobsOptions => ListBlobHierarchyArgs)
       : ListService[F] =
     toListBlobsOption map
       mkArgs andThen
       requests.listRequestK andThen
-      converters.toBlobstorePathsK
+      converters.toBlobstorePathsK.map(_.some)
```
Comment (rintcius) [inline azure/src/main/scala/quasar/blobstore/azure/AzureListService.scala:42]: As discussed with @wemrysi we are deviating from the datasource spec here. The spec mandates to return `None` for non-existing paths and `Some` (empty stream) for existing non-prefix blobpaths. However we are returning `Some` empty stream for both cases here.

So no need to use `peek1` anymore. 

Created ch11270 to discuss further

**Comment by rintcius** [comment]
Published snapshot `2.1.13-bd49bf9`

**Review by wemrysi — APPROVED** [review]
One minor comment regarding the tests, you don't need to wrap all the examples in `IO`. I'd argue that doing so creates a bit of confusion since it isn't necessary.

**Comment by beckyconning** [comment]
Fails immediately and silently when blob name already exists. S3 behaviour is to overwrite.

**Pull request merged**